rsocket-core/src/main/java/io/rsocket/internal/BaseDuplexConnection.java

```diff
@@ -11,7 +11,7 @@
   protected UnboundedProcessor sender = new UnboundedProcessor();
 
   public BaseDuplexConnection() {
-    onClose.doFinally(s -> doOnClose()).subscribe();
+    onClose.subscribe(null, t -> doOnClose(), this::doOnClose);
```

Comment:
unrelated, non-blocking.  What's the plan with the deprecation? https://github.com/reactor/reactor-core/issues/2431#issuecomment-706324683

Should onClose be updated?

Reply:
Yeah. I will do that in a separate PR